pkg/agent/run.go

```diff
@@ -40,91 +43,98 @@ var Flags AgentCmdFlags
 	defer cancel()
 
+	log.Info("starting", "version", version.PreflightVersion, "commit", version.Commit)
+
 	file, err := os.Open(Flags.ConfigFilePath)
 	if err != nil {
-		logs.Log.Fatalf("Failed to load config file for agent from: %s", Flags.ConfigFilePath)
+		return fmt.Errorf("Failed to load config file for agent from %q: %s", Flags.ConfigFilePath, err)
 	}
 	defer file.Close()
 
 	b, err := io.ReadAll(file)
 	if err != nil {
-		logs.Log.Fatalf("Failed to read config file: %s", err)
+		return fmt.Errorf("Failed to read config file: %s", err)
 	}
 
 	cfg, err := ParseConfig(b)
 	if err != nil {
-		logs.Log.Fatalf("Failed to parse config file: %s", err)
+		return fmt.Errorf("Failed to parse config file: %s", err)
 	}
 
-	config, preflightClient, err := ValidateAndCombineConfig(logs.Log, cfg, Flags)
+	config, preflightClient, err := ValidateAndCombineConfig(log, cfg, Flags)
 	if err != nil {
-		logs.Log.Fatalf("While evaluating configuration: %v", err)
+		return fmt.Errorf("Failed to evaluate configuration: %s", err)
 	}
 
+	serverMux := http.NewServeMux()
+
 	if Flags.Profiling {
-		logs.Log.Printf("pprof profiling was enabled.\nRunning profiling on port :6060")
-		go func() {
-			err := http.ListenAndServe(":6060", nil)
-			if err != nil && !errors.Is(err, http.ErrServerClosed) {
-				logs.Log.Fatalf("failed to run pprof profiler: %s", err)
-			}
-		}()
-	}
+		log.Info("pprof profiling enabled")
+		serverMux.HandleFunc("/debug/pprof/", pprof.Index)
+		serverMux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+		serverMux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+		serverMux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+		serverMux.HandleFunc("/debug/pprof/trace", pprof.Trace)
 
-	go func() {
-		server := http.NewServeMux()
+	}
 
-		if Flags.Prometheus {
-			logs.Log.Printf("Prometheus was enabled.\nRunning prometheus on port :8081")
-			prometheus.MustRegister(metricPayloadSize)
-			server.Handle("/metrics", promhttp.Handler())
-		}
+	if Flags.Prometheus {
+		log.Info("Prometheus metrics enabled")
+		prometheus.MustRegister(metricPayloadSize)
+		serverMux.Handle("/metrics", promhttp.Handler())
+	}
 
-		// Health check endpoint. Since we haven't figured a good way of knowning
-		// what "ready" means for the agent, we just return 200 OK inconditionally.
-		// The goal is to satisfy some Kubernetes distributions, like OpenShift,
-		// that require a liveness and health probe to be present for each pod.
-		server.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
-			w.WriteHeader(http.StatusOK)
-		})
-		server.HandleFunc("/readyz", func(w http.ResponseWriter, r *http.Request) {
-			w.WriteHeader(http.StatusOK)
+	// Health check endpoint. Since we haven't figured a good way of knowning
+	// what "ready" means for the agent, we just return 200 OK inconditionally.
+	// The goal is to satisfy some Kubernetes distributions, like OpenShift,
+	// that require a liveness and health probe to be present for each pod.
```

Comment:
Not sure what "health" probe means here.
I thought that only liveness probe was required by OpenShift?
ReadinessProbe makes no sense for venafi-kubernetes-agent.

Reply:
My mistake, I confused "health probe" with "readiness probe". It should say "readiness probe" here.